lib/src/main/java/io/ably/lib/types/ProtocolMessage.java

```diff
@@ -205,4 +214,42 @@ public JsonElement serialize(Action action, Type t, JsonSerializationContext ctx
 			return new JsonPrimitive(action.getValue());
 		}		
 	}
+
+	public static class AuthDetails {
+		public String accessToken;
+
+		private AuthDetails() { }
+		public AuthDetails(String s) { accessToken = s; }
+
+		AuthDetails readMsgpack(MessageUnpacker unpacker) throws IOException {
+			int fieldCount = unpacker.unpackMapHeader();
+			for(int i = 0; i < fieldCount; i++) {
+				String fieldName = unpacker.unpackString().intern();
+				MessageFormat fieldFormat = unpacker.getNextFormat();
+				if(fieldFormat.equals(MessageFormat.NIL)) { unpacker.unpackNil(); continue; }
+
+				if(fieldName == "accessToken") {
+					accessToken = unpacker.unpackString();
+				} else {
+					System.out.println("Unexpected field: " + fieldName);
```

Comment:
Yes, but my point is that it should not be going to system out ever anywhere in the library, it should be going to the logger with a suitable log severity.  I am not bothered by which severity you think is appropriate, but I don't like that our library code (not test code) is incorrectly writing to STDOUT. Can we fix anywhere else this happens?